### PR TITLE
TASK-106 Add project roadmap milestones

### PR DESCRIPTION
**Pull request by dorianagaesse**
## Summary
- add a dedicated `Roadmap` section with standalone project milestones and creative desktop/mobile timeline layouts
- add roadmap milestone schema, migration, service layer, and human-only project API routes including ordering
- cover the rollout with roadmap service, API, and component tests and update project tracking/docs

## Validation
- `npx -y -p node@20.19.0 node .\\node_modules\\prisma\\build\\index.js generate`
- `npx -y -p node@20.19.0 node .\\node_modules\\eslint\\bin\\eslint.js .`
- `npx -y -p node@20.19.0 node .\\node_modules\\vitest\\vitest.mjs run tests\\lib\\project-roadmap-service.test.ts tests\\api\\project-roadmap.route.test.ts tests\\components\\project-roadmap-panel.test.tsx`
- `DATABASE_URL=... DIRECT_URL=... AGENT_TOKEN_SIGNING_SECRET=... RESEND_API_KEY=... npx -y -p node@20.19.0 node .\\node_modules\\vitest\\vitest.mjs run`
- `DATABASE_URL=... DIRECT_URL=... AGENT_TOKEN_SIGNING_SECRET=... RESEND_API_KEY=... GOOGLE_TOKEN_ENCRYPTION_KEY=... npx -y -p node@20.19.0 node .\\node_modules\\next\\dist\\bin\\next build`

### CODE REVIEW

File: lib/services/project-roadmap-service.ts [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+  return milestoneIds.every((milestoneId) => {
+    if (typeof milestoneId !== "string" || milestoneId.trim().length === 0) {
+      return false;
+    }
+
+    if (seenMilestoneIds.has(milestoneId)) {
+      return false;
+    }
+
+    seenMilestoneIds.add(milestoneId);
```
[the remark this answers is not in the document]
Comment (dorianagaesse) [reply lib/services/project-roadmap-service.ts]: Fixed in `7e56c21`: reorder validation now rejects IDs with surrounding whitespace and deduplicates on the trimmed value. I also added a regression test for the whitespace case.

File: app/api/projects/[projectId]/roadmap-milestones/[milestoneId]/route.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          description:
+            typeof payload.description === "string" ? payload.description : null,
+        }
+      : {}),
+    ...(Object.prototype.hasOwnProperty.call(payload, "targetDate")
+      ? {
+          targetDate:
+            typeof payload.targetDate === "string" ? payload.targetDate : null,
+        }
+      : {}),
```
[the remark this answers is not in the document]
Comment (dorianagaesse) [reply app/api/projects/[projectId]/roadmap-milestones/[milestoneId]/route.ts]: Fixed in `7e56c21`: the PATCH route now returns `400 invalid-payload` when `title` is non-string or when `description` / `targetDate` / `status` are present with unsupported types, instead of coercing malformed values into destructive clears. Added route coverage for the invalid-description case.

**Comment by dorianagaesse** [comment]
Closing this PR as superseded by #198.

`feature/task-130-roadmap-groups` already contains the full `TASK-106` roadmap foundation and the final validated roadmap v2 work that should land in `main`, so keeping both PRs open would create a redundant merge path.
